x/programs/runtime/runtime.go

```diff
@@ -123,7 +126,7 @@ func (r *runtime) Call(ctx context.Context, name string, params ...uint64) ([]ui
 	}
 
 	if api == nil {
```

Comment:
as pre comment above this should never happen if we properly register.

Reply:
so previously we passed in a list of function names, each that mapped to an `api.function`. I think it simplifies the flow if we just grab the `api_function` when we need it(this would be in `call`).  That way we don't have to store the map, users don't need to pass in function names, and call becomes much more modular.

Reply:
Yeah I guess once the mod is initialized the work here is already done.